service/src/test/java/bio/terra/pipelines/service/PipelinesServiceTest.java

```diff
@@ -170,4 +195,162 @@ void updatePipelineWorkspaceId() {
+        arguments(PipelineInputTypesEnum.STRING, "  I am a string  ", true),
+        arguments(
+            PipelineInputTypesEnum.STRING, List.of("this", "is", "not", "a", "string"), false),
+        arguments(PipelineInputTypesEnum.STRING, 123, false),
+        arguments(PipelineInputTypesEnum.STRING, null, false),
+        arguments(PipelineInputTypesEnum.STRING, "", false),
+
+        // VCF
+        arguments(PipelineInputTypesEnum.VCF, "path/to/file.vcf.gz", true),
+        arguments(PipelineInputTypesEnum.VCF, "path/to/file.vcf", false),
+        arguments(PipelineInputTypesEnum.VCF, 3, false),
+        arguments(PipelineInputTypesEnum.VCF, null, false),
+        arguments(PipelineInputTypesEnum.VCF, "", false),
+
+        // STRING_ARRAY
+        arguments(
+            PipelineInputTypesEnum.STRING_ARRAY,
+            Arrays.asList("this", "is", "a", "list", "of", "strings"),
+            true),
+        arguments(
+            PipelineInputTypesEnum.STRING_ARRAY,
+            Arrays.asList("this ", " is", " a ", "list  ", "  of", "  strings  "),
+            true),
+        arguments(PipelineInputTypesEnum.STRING_ARRAY, "I am not an array", false),
+        arguments(PipelineInputTypesEnum.STRING_ARRAY, Arrays.asList(1, 2, 3), false),
+        arguments(PipelineInputTypesEnum.STRING_ARRAY, null, false),
+        arguments(PipelineInputTypesEnum.STRING_ARRAY, List.of(), false),
+        arguments(
+            PipelineInputTypesEnum.STRING_ARRAY,
+            Arrays.asList("", "array with empty string"),
+            false),
+        arguments(
+            PipelineInputTypesEnum.STRING_ARRAY, Arrays.asList(null, "array with null"), false),
+
+        // VCF_ARRAY
+        arguments(PipelineInputTypesEnum.VCF_ARRAY, List.of("path/to/file.vcf.gz"), true),
+        arguments(PipelineInputTypesEnum.VCF_ARRAY, List.of(" path/to/file.vcf.gz  "), true),
+        arguments(PipelineInputTypesEnum.VCF_ARRAY, "this/is/not/an/array.vcf.gz", false),
+        arguments(
+            PipelineInputTypesEnum.VCF_ARRAY,
+            Arrays.asList("path/to/file.vcf.gz", "not a path"),
+            false),
+        arguments(PipelineInputTypesEnum.VCF_ARRAY, null, false),
+        arguments(PipelineInputTypesEnum.VCF_ARRAY, List.of(), false),
+        arguments(
+            PipelineInputTypesEnum.VCF_ARRAY, Arrays.asList(null, "list/with/null.vcf.gz"), false));
+  }
+
+  @ParameterizedTest
+  @MethodSource("inputTypeValidations")
+  void validateInputType(
+      PipelineInputTypesEnum inputType, Object inputValue, Boolean shouldPassValidation) {
+    PipelineInputDefinition inputDefinition =
+        new PipelineInputDefinition(1L, "input_name", inputType.toString(), true);
+    List<PipelineInputDefinition> inputDefinitions = new ArrayList<>(List.of(inputDefinition));
+
+    LinkedHashMap<String, Object> inputs = new LinkedHashMap<>();
+    inputs.put("input_name", inputValue);
+
+    if (shouldPassValidation) {
```

Comment:
i think you can change this block to 

```
assertEquals(shouldPassValidation, PipelinesService.validateInputTypes(inputDefinitions, inputs).isEmpty())
```

unless you're testing only 1 error gets returned as well

Reply:
oh nice, thanks!